src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,162 @@
+
+    getValue() {
+        if (this.state.editor) {
+            return this.state.editor.getValue();
+        }
+    }
+
+    setValue(newValue) {
+        if (this.state.editor) {
+            this.state.editor.setValue(newValue);
+        }
+    }
+
+    setReadOnly(readOnly) {
+        if (this.state.editor) {
+            this.setState({ readOnly: readOnly });
+        }
+    }
+
+    setError(error) {
+        if (!this.state.editor || !error) {
+            return;
+        }
+        monaco.init().then((monaco) => {
+            this.state.decorations = this.state.editor
+                .getModel()
+                .deltaDecorations(
+                    [],
+                    [
+                        {
+                            range: new monaco.Range(
+                                error[1],
+                                error[2],
+                                error[1],
+                                error[3] + 1 // Increase by one because it excludes end column
+                            ),
+                            options: {
+                                isWholeLine: error.length > 3 ? false : true,
+                                className: "contentError",
+                                glyphMarginClassName: "glyphMargin",
+                            },
+                        },
+                    ]
+                );
+        });
+    }
+
+    handleEditorDidMount(_, editor) {
+        this.setState({ editor, editor });
+
+        monaco
+            .init()
+            .then((monaco) => {
+                // Register a new yara language
+                monaco.languages.register({ id: "yara" });
+
+                // Register a tokens provider for yara
+                monaco.languages.setMonarchTokensProvider(
+                    "yara",
+                    YARA.TOKEN_PROVIDER
+                );
+
+                // Register a completion item provider for yara
+                monaco.languages.registerCompletionItemProvider("yara", {
+                    provideCompletionItems: () => {
+                        var suggestions = [
+                            {
+                                label: "rule",
+                                kind:
+                                    monaco.languages.CompletionItemKind.Snippet,
+                                insertText: YARA.COMPLETION_RULE,
+                                insertTextRules:
+                                    monaco.languages
+                                        .CompletionItemInsertTextRule
+                                        .InsertAsSnippet,
+                                documentation: "Generate a rule skeleton",
+                            },
+                        ];
+                        return { suggestions: suggestions };
+                    },
+                });
+
+                // We want to have a wa to indicate the user that the
+                // Editor is currently in read-only mode
```

Comment:
```suggestion
                // We want to have a way to indicate the user that the
                // editor is currently in read-only mode
```

Reply:
`Editor` is the class of the object. Otherwise I'd write "that **the editor**"

Reply:
I just saw the typo in a `way`, the `Editor` one was a nitpick (but right, it's correct like it is)